src/connection.rs

```diff
@@ -154,11 +154,20 @@ impl IntoConnectionInfo for url::Url {
     }
 }
 
+struct TcpConnection {
+    pub reader: BufReader<TcpStream>,
```

Comment:
Do these have to be public? It's all internal in this module, isn't it?

Reply:
You are right, they don't.